leakcanary-support-fragment/src/androidTest/java/com/squareup/leakcanary/tests/Fragments.kt

```diff
@@ -0,0 +1,49 @@
+package com.squareup.leakcanary.tests
+
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentActivity
+import androidx.fragment.app.FragmentManager
+import java.util.concurrent.CountDownLatch
+
+class Fragments private constructor() {
+
+  init {
+    throw AssertionError()
```

Comment:
This would throw an `AssertionError` every time a `Fragments` class is create.

For this class, you would want to convert the `waitForFragmentDetached` and `waitForFragmentViewDestroyed` into extension functions on `FragmentActivity` which returns a `CountDownLatch`.

Something like `internal fun FragmentActivity.waitForFragmentViewDestroyed(): CountDownLatch`

Reply:
This was original behaviour from the Java code, however no longer necessary with the extension function approach 👍